rpmlint/checks/BinariesCheck.py

```diff
@@ -389,6 +406,10 @@ def check_binary(self, pkg):
             is_shlib = so_regex.search(fname)
             bin_info = BinaryInfo(self.config, self.output, pkg, pkgfile.path, fname, is_ar, is_shlib)
 
+            bin_info.checker.check()
```

Comment:
And here just don't bother with the bin_info and directly do the work yourself. Instead of having another class doing it for you.